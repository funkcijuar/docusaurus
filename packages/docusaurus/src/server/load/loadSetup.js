/**
 * Copyright (c) 2017-present, Facebook, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import path from 'path';
import load from '../index';

// Helper methods to setup dummy/fake projects
const loadSetup = async name => {
  const fixtures = path.join(__dirname, '__tests__', '__fixtures__');
  const simpleSite = path.join(fixtures, 'simple-site');
  const customSite = path.join(fixtures, 'custom-site');

  switch (name) {
    case 'simple':
      return load(simpleSite);
    case 'custom':
      return load(customSite);
    default:
      return {};
  }
};

export default loadSetup;
