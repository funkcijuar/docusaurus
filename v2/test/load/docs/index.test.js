import loadDocs from '@lib/load/docs';
import loadSetup from '../../loadSetup';

describe('loadDocs', () => {
  test('simple website', async () => {
    const props = await loadSetup('simple');
    const {siteDir, docsDir, env, siteConfig} = props;
    const {docsMetadatas} = await loadDocs({siteDir, docsDir, env, siteConfig});
    expect(docsMetadatas).toMatchSnapshot();
  });

  test('versioned website', async () => {
    const props = await loadSetup('versioned');
    const {siteDir, docsDir, env, siteConfig} = props;
    const {docsMetadatas} = await loadDocs({siteDir, docsDir, env, siteConfig});
    expect(docsMetadatas).toMatchSnapshot();
  });

  test('versioned & translated website', async () => {
    const props = await loadSetup('transversioned');
    const {siteDir, docsDir, env, siteConfig} = props;
    const {docsMetadatas} = await loadDocs({siteDir, docsDir, env, siteConfig});
    expect(docsMetadatas).toMatchSnapshot();
  });

  test('translated website', async () => {
    const props = await loadSetup('translated');
    const {siteDir, docsDir, env, siteConfig} = props;
    const {docsMetadatas} = await loadDocs({siteDir, docsDir, env, siteConfig});
    expect(docsMetadatas).toMatchSnapshot();
  });
});
