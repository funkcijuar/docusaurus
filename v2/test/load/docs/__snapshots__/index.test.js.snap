// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadDocs simple website 1`] = `
Object {
  "foo/bar": Object {
    "category": "Test",
    "id": "foo/bar",
    "language": undefined,
    "localized_id": "foo/bar",
    "next": "foo/baz",
    "next_id": "foo/baz",
    "next_title": "baz",
    "permalink": "/docs/foo/bar",
    "sidebar": "docs",
    "source": "@docs/foo/bar.md",
    "title": "Bar",
    "version": undefined,
  },
  "foo/baz": Object {
    "category": "Test",
    "id": "foo/baz",
    "language": undefined,
    "localized_id": "foo/baz",
    "next": "hello",
    "next_id": "hello",
    "next_title": "Hello, World !",
    "permalink": "/docs/foo/baz",
    "previous": "foo/bar",
    "previous_id": "foo/bar",
    "previous_title": "Bar",
    "sidebar": "docs",
    "source": "@docs/foo/baz.md",
    "title": "baz",
    "version": undefined,
  },
  "hello": Object {
    "category": "Guides",
    "id": "hello",
    "language": undefined,
    "localized_id": "hello",
    "permalink": "/docs/hello",
    "previous": "foo/baz",
    "previous_id": "foo/baz",
    "previous_title": "baz",
    "sidebar": "docs",
    "source": "@docs/hello.md",
    "title": "Hello, World !",
    "version": undefined,
  },
  "permalink": Object {
    "id": "permalink",
    "language": undefined,
    "localized_id": "permalink",
    "permalink": "/docs/endiliey/permalink",
    "source": "@docs/permalink.md",
    "title": "Permalink",
    "version": undefined,
  },
}
`;

exports[`loadDocs translated website 1`] = `
Object {
  "en-foo/bar": Object {
    "category": "Test",
    "id": "en-foo/bar",
    "language": "en",
    "localized_id": "foo/bar",
    "next": "en-foo/baz",
    "next_id": "foo/baz",
    "next_title": "baz",
    "permalink": "/docs/en/foo/bar",
    "sidebar": "docs",
    "source": "@docs/foo/bar.md",
    "title": "Bar",
    "version": undefined,
  },
  "en-foo/baz": Object {
    "category": "Test",
    "id": "en-foo/baz",
    "language": "en",
    "localized_id": "foo/baz",
    "next": "en-hello",
    "next_id": "hello",
    "next_title": "Hello, World !",
    "permalink": "/docs/en/foo/baz",
    "previous": "en-foo/bar",
    "previous_id": "foo/bar",
    "previous_title": "Bar",
    "sidebar": "docs",
    "source": "@docs/foo/baz.md",
    "title": "baz",
    "version": undefined,
  },
  "en-hello": Object {
    "category": "Guides",
    "id": "en-hello",
    "language": "en",
    "localized_id": "hello",
    "permalink": "/docs/en/hello",
    "previous": "en-foo/baz",
    "previous_id": "foo/baz",
    "previous_title": "baz",
    "sidebar": "docs",
    "source": "@docs/hello.md",
    "title": "Hello, World !",
    "version": undefined,
  },
  "en-permalink": Object {
    "id": "en-permalink",
    "language": "en",
    "localized_id": "permalink",
    "permalink": "/docs/en/endiliey/permalink",
    "source": "@docs/permalink.md",
    "title": "Permalink",
    "version": undefined,
  },
  "ko-foo/bar": Object {
    "category": "Test",
    "id": "ko-foo/bar",
    "language": "ko",
    "localized_id": "foo/bar",
    "next": "ko-foo/baz",
    "next_id": "foo/baz",
    "next_title": "baz",
    "permalink": "/docs/ko/foo/bar",
    "sidebar": "docs",
    "source": "@translated_docs/ko/foo/bar.md",
    "title": "Bar",
    "version": undefined,
  },
  "ko-foo/baz": Object {
    "category": "Test",
    "id": "ko-foo/baz",
    "language": "ko",
    "localized_id": "foo/baz",
    "next": "ko-hello",
    "next_id": "hello",
    "next_title": "Hello, World !",
    "permalink": "/docs/ko/foo/baz",
    "previous": "ko-foo/bar",
    "previous_id": "foo/bar",
    "previous_title": "Bar",
    "sidebar": "docs",
    "source": "@translated_docs/ko/foo/baz.md",
    "title": "baz",
    "version": undefined,
  },
  "ko-hello": Object {
    "category": "Guides",
    "id": "ko-hello",
    "language": "ko",
    "localized_id": "hello",
    "permalink": "/docs/ko/hello",
    "previous": "ko-foo/baz",
    "previous_id": "foo/baz",
    "previous_title": "baz",
    "sidebar": "docs",
    "source": "@translated_docs/ko/hello.md",
    "title": "Hello, World !",
    "version": undefined,
  },
}
`;

exports[`loadDocs versioned & translated website 1`] = `
Object {
  "en-foo/bar": Object {
    "category": "Test",
    "id": "en-foo/bar",
    "language": "en",
    "localized_id": "foo/bar",
    "next": "en-foo/baz",
    "next_id": "foo/baz",
    "next_title": "baz",
    "permalink": "/docs/en/next/foo/bar",
    "sidebar": "docs",
    "source": "@docs/foo/bar.md",
    "title": "Bar",
    "version": "next",
  },
  "en-foo/baz": Object {
    "category": "Test",
    "id": "en-foo/baz",
    "language": "en",
    "localized_id": "foo/baz",
    "next": "en-hello",
    "next_id": "hello",
    "next_title": "Hello, World !",
    "permalink": "/docs/en/next/foo/baz",
    "previous": "en-foo/bar",
    "previous_id": "foo/bar",
    "previous_title": "Bar",
    "sidebar": "docs",
    "source": "@docs/foo/baz.md",
    "title": "baz",
    "version": "next",
  },
  "en-hello": Object {
    "category": "Guides",
    "id": "en-hello",
    "language": "en",
    "localized_id": "hello",
    "permalink": "/docs/en/next/hello",
    "previous": "en-foo/baz",
    "previous_id": "foo/baz",
    "previous_title": "baz",
    "sidebar": "docs",
    "source": "@docs/hello.md",
    "title": "Hello, World !",
    "version": "next",
  },
  "en-permalink": Object {
    "id": "en-permalink",
    "language": "en",
    "localized_id": "permalink",
    "permalink": "/docs/en/next/endiliey/permalink",
    "source": "@docs/permalink.md",
    "title": "Permalink",
    "version": "next",
  },
  "en-version-1.0.0-foo/bar": Object {
    "category": "Test",
    "id": "en-version-1.0.0-foo/bar",
    "language": "en",
    "localized_id": "version-1.0.0-foo/bar",
    "next": "en-version-1.0.0-foo/baz",
    "next_id": "version-1.0.0-foo/baz",
    "next_title": "Baz",
    "permalink": "/docs/en/1.0.0/foo/bar",
    "sidebar": "version-1.0.0-docs",
    "source": "@versioned_docs/version-1.0.0/foo/bar.md",
    "title": "Bar",
    "version": "1.0.0",
  },
  "en-version-1.0.0-foo/baz": Object {
    "category": "Test",
    "id": "en-version-1.0.0-foo/baz",
    "language": "en",
    "localized_id": "version-1.0.0-foo/baz",
    "next": "en-version-1.0.0-hello",
    "next_id": "version-1.0.0-hello",
    "next_title": "Hello, World !",
    "permalink": "/docs/en/1.0.0/foo/baz",
    "previous": "en-version-1.0.0-foo/bar",
    "previous_id": "version-1.0.0-foo/bar",
    "previous_title": "Bar",
    "sidebar": "version-1.0.0-docs",
    "source": "@versioned_docs/version-1.0.0/foo/baz.md",
    "title": "Baz",
    "version": "1.0.0",
  },
  "en-version-1.0.0-hello": Object {
    "category": "Guides",
    "id": "en-version-1.0.0-hello",
    "language": "en",
    "localized_id": "version-1.0.0-hello",
    "permalink": "/docs/en/1.0.0/hello",
    "previous": "en-version-1.0.0-foo/baz",
    "previous_id": "version-1.0.0-foo/baz",
    "previous_title": "Baz",
    "sidebar": "version-1.0.0-docs",
    "source": "@versioned_docs/version-1.0.0/hello.md",
    "title": "Hello, World !",
    "version": "1.0.0",
  },
  "en-version-1.0.1-foo/bar": Object {
    "category": "Test",
    "id": "en-version-1.0.1-foo/bar",
    "language": "en",
    "localized_id": "version-1.0.1-foo/bar",
    "next": "en-version-1.0.1-foo/baz",
    "next_id": "version-1.0.1-foo/baz",
    "next_title": "Baz",
    "permalink": "/docs/en/foo/bar",
    "sidebar": "version-1.0.1-docs",
    "source": "@versioned_docs/version-1.0.1/foo/bar.md",
    "title": "Bar",
    "version": "1.0.1",
  },
  "en-version-1.0.1-foo/baz": Object {
    "category": "Test",
    "id": "en-version-1.0.1-foo/baz",
    "language": "en",
    "localized_id": "version-1.0.1-foo/baz",
    "next": "en-version-1.0.1-hello",
    "next_id": "version-1.0.1-hello",
    "next_title": "Hello, World !",
    "permalink": "/docs/en/foo/baz",
    "previous": "en-version-1.0.1-foo/bar",
    "previous_id": "version-1.0.1-foo/bar",
    "previous_title": "Bar",
    "sidebar": "version-1.0.1-docs",
    "source": "@versioned_docs/version-1.0.1/foo/baz.md",
    "title": "Baz",
    "version": "1.0.1",
  },
  "en-version-1.0.1-hello": Object {
    "category": "Guides",
    "id": "en-version-1.0.1-hello",
    "language": "en",
    "localized_id": "version-1.0.1-hello",
    "permalink": "/docs/en/hello",
    "previous": "en-version-1.0.1-foo/baz",
    "previous_id": "version-1.0.1-foo/baz",
    "previous_title": "Baz",
    "sidebar": "version-1.0.1-docs",
    "source": "@versioned_docs/version-1.0.1/hello.md",
    "title": "Hello, World !",
    "version": "1.0.1",
  },
  "ko-foo/bar": Object {
    "category": "Test",
    "id": "ko-foo/bar",
    "language": "ko",
    "localized_id": "foo/bar",
    "next": "ko-foo/baz",
    "next_id": "foo/baz",
    "next_title": "baz",
    "permalink": "/docs/ko/next/foo/bar",
    "sidebar": "docs",
    "source": "@translated_docs/ko/foo/bar.md",
    "title": "Bar",
    "version": "next",
  },
  "ko-foo/baz": Object {
    "category": "Test",
    "id": "ko-foo/baz",
    "language": "ko",
    "localized_id": "foo/baz",
    "next": "ko-hello",
    "next_id": "hello",
    "next_title": "Hello, World !",
    "permalink": "/docs/ko/next/foo/baz",
    "previous": "ko-foo/bar",
    "previous_id": "foo/bar",
    "previous_title": "Bar",
    "sidebar": "docs",
    "source": "@translated_docs/ko/foo/baz.md",
    "title": "baz",
    "version": "next",
  },
  "ko-hello": Object {
    "category": "Guides",
    "id": "ko-hello",
    "language": "ko",
    "localized_id": "hello",
    "permalink": "/docs/ko/next/hello",
    "previous": "ko-foo/baz",
    "previous_id": "foo/baz",
    "previous_title": "baz",
    "sidebar": "docs",
    "source": "@translated_docs/ko/hello.md",
    "title": "Hello, World !",
    "version": "next",
  },
  "ko-version-1.0.0-foo/bar": Object {
    "category": "Test",
    "id": "ko-version-1.0.0-foo/bar",
    "language": "ko",
    "localized_id": "version-1.0.0-foo/bar",
    "next": "ko-version-1.0.0-foo/baz",
    "next_id": "version-1.0.0-foo/baz",
    "next_title": "baz",
    "permalink": "/docs/ko/1.0.0/foo/bar",
    "sidebar": "version-1.0.0-docs",
    "source": "@translated_docs/ko/version-1.0.0/foo/bar.md",
    "title": "Bar",
    "version": "1.0.0",
  },
  "ko-version-1.0.0-foo/baz": Object {
    "category": "Test",
    "id": "ko-version-1.0.0-foo/baz",
    "language": "ko",
    "localized_id": "version-1.0.0-foo/baz",
    "next": "ko-version-1.0.0-hello",
    "next_id": "version-1.0.0-hello",
    "next_title": "Hello, World !",
    "permalink": "/docs/ko/1.0.0/foo/baz",
    "previous": "ko-version-1.0.0-foo/bar",
    "previous_id": "version-1.0.0-foo/bar",
    "previous_title": "Bar",
    "sidebar": "version-1.0.0-docs",
    "source": "@translated_docs/ko/version-1.0.0/foo/baz.md",
    "title": "baz",
    "version": "1.0.0",
  },
  "ko-version-1.0.0-hello": Object {
    "category": "Guides",
    "id": "ko-version-1.0.0-hello",
    "language": "ko",
    "localized_id": "version-1.0.0-hello",
    "permalink": "/docs/ko/1.0.0/hello",
    "previous": "ko-version-1.0.0-foo/baz",
    "previous_id": "version-1.0.0-foo/baz",
    "previous_title": "baz",
    "sidebar": "version-1.0.0-docs",
    "source": "@translated_docs/ko/version-1.0.0/hello.md",
    "title": "Hello, World !",
    "version": "1.0.0",
  },
  "ko-version-1.0.1-foo/bar": Object {
    "category": "Test",
    "id": "ko-version-1.0.1-foo/bar",
    "language": "ko",
    "localized_id": "version-1.0.1-foo/bar",
    "next": "ko-version-1.0.1-foo/baz",
    "next_id": "version-1.0.1-foo/baz",
    "next_title": "baz",
    "permalink": "/docs/ko/foo/bar",
    "sidebar": "version-1.0.1-docs",
    "source": "@translated_docs/ko/version-1.0.1/foo/bar.md",
    "title": "Bar",
    "version": "1.0.1",
  },
  "ko-version-1.0.1-foo/baz": Object {
    "category": "Test",
    "id": "ko-version-1.0.1-foo/baz",
    "language": "ko",
    "localized_id": "version-1.0.1-foo/baz",
    "next": "ko-version-1.0.1-hello",
    "next_id": "version-1.0.1-hello",
    "next_title": "Hello, World !",
    "permalink": "/docs/ko/foo/baz",
    "previous": "ko-version-1.0.1-foo/bar",
    "previous_id": "version-1.0.1-foo/bar",
    "previous_title": "Bar",
    "sidebar": "version-1.0.1-docs",
    "source": "@translated_docs/ko/version-1.0.1/foo/baz.md",
    "title": "baz",
    "version": "1.0.1",
  },
  "ko-version-1.0.1-hello": Object {
    "category": "Guides",
    "id": "ko-version-1.0.1-hello",
    "language": "ko",
    "localized_id": "version-1.0.1-hello",
    "permalink": "/docs/ko/hello",
    "previous": "ko-version-1.0.1-foo/baz",
    "previous_id": "version-1.0.1-foo/baz",
    "previous_title": "baz",
    "sidebar": "version-1.0.1-docs",
    "source": "@translated_docs/ko/version-1.0.1/hello.md",
    "title": "Hello, World !",
    "version": "1.0.1",
  },
}
`;

exports[`loadDocs versioned website 1`] = `
Object {
  "foo/bar": Object {
    "category": "Test",
    "id": "foo/bar",
    "language": undefined,
    "localized_id": "foo/bar",
    "next": "foo/baz",
    "next_id": "foo/baz",
    "next_title": "baz",
    "permalink": "/docs/next/foo/bar",
    "sidebar": "docs",
    "source": "@docs/foo/bar.md",
    "title": "Bar",
    "version": "next",
  },
  "foo/baz": Object {
    "category": "Test",
    "id": "foo/baz",
    "language": undefined,
    "localized_id": "foo/baz",
    "next": "hello",
    "next_id": "hello",
    "next_title": "Hello, World !",
    "permalink": "/docs/next/foo/baz",
    "previous": "foo/bar",
    "previous_id": "foo/bar",
    "previous_title": "Bar",
    "sidebar": "docs",
    "source": "@docs/foo/baz.md",
    "title": "baz",
    "version": "next",
  },
  "hello": Object {
    "category": "Guides",
    "id": "hello",
    "language": undefined,
    "localized_id": "hello",
    "permalink": "/docs/next/hello",
    "previous": "foo/baz",
    "previous_id": "foo/baz",
    "previous_title": "baz",
    "sidebar": "docs",
    "source": "@docs/hello.md",
    "title": "Hello, World !",
    "version": "next",
  },
  "permalink": Object {
    "id": "permalink",
    "language": undefined,
    "localized_id": "permalink",
    "permalink": "/docs/next/endiliey/permalink",
    "source": "@docs/permalink.md",
    "title": "Permalink",
    "version": "next",
  },
  "version-1.0.0-foo/bar": Object {
    "category": "Test",
    "id": "version-1.0.0-foo/bar",
    "language": undefined,
    "localized_id": "version-1.0.0-foo/bar",
    "next": "version-1.0.0-foo/baz",
    "next_id": "version-1.0.0-foo/baz",
    "next_title": "Baz",
    "permalink": "/docs/1.0.0/foo/bar",
    "sidebar": "version-1.0.0-docs",
    "source": "@versioned_docs/version-1.0.0/foo/bar.md",
    "title": "Bar",
    "version": "1.0.0",
  },
  "version-1.0.0-foo/baz": Object {
    "category": "Test",
    "id": "version-1.0.0-foo/baz",
    "language": undefined,
    "localized_id": "version-1.0.0-foo/baz",
    "next": "version-1.0.0-hello",
    "next_id": "version-1.0.0-hello",
    "next_title": "Hello, World !",
    "permalink": "/docs/1.0.0/foo/baz",
    "previous": "version-1.0.0-foo/bar",
    "previous_id": "version-1.0.0-foo/bar",
    "previous_title": "Bar",
    "sidebar": "version-1.0.0-docs",
    "source": "@versioned_docs/version-1.0.0/foo/baz.md",
    "title": "Baz",
    "version": "1.0.0",
  },
  "version-1.0.0-hello": Object {
    "category": "Guides",
    "id": "version-1.0.0-hello",
    "language": undefined,
    "localized_id": "version-1.0.0-hello",
    "permalink": "/docs/1.0.0/hello",
    "previous": "version-1.0.0-foo/baz",
    "previous_id": "version-1.0.0-foo/baz",
    "previous_title": "Baz",
    "sidebar": "version-1.0.0-docs",
    "source": "@versioned_docs/version-1.0.0/hello.md",
    "title": "Hello, World !",
    "version": "1.0.0",
  },
  "version-1.0.1-foo/bar": Object {
    "category": "Test",
    "id": "version-1.0.1-foo/bar",
    "language": undefined,
    "localized_id": "version-1.0.1-foo/bar",
    "next": "version-1.0.1-foo/baz",
    "next_id": "version-1.0.1-foo/baz",
    "next_title": "Baz",
    "permalink": "/docs/foo/bar",
    "sidebar": "version-1.0.1-docs",
    "source": "@versioned_docs/version-1.0.1/foo/bar.md",
    "title": "Bar",
    "version": "1.0.1",
  },
  "version-1.0.1-foo/baz": Object {
    "category": "Test",
    "id": "version-1.0.1-foo/baz",
    "language": undefined,
    "localized_id": "version-1.0.1-foo/baz",
    "next": "version-1.0.1-hello",
    "next_id": "version-1.0.1-hello",
    "next_title": "Hello, World !",
    "permalink": "/docs/foo/baz",
    "previous": "version-1.0.1-foo/bar",
    "previous_id": "version-1.0.1-foo/bar",
    "previous_title": "Bar",
    "sidebar": "version-1.0.1-docs",
    "source": "@versioned_docs/version-1.0.1/foo/baz.md",
    "title": "Baz",
    "version": "1.0.1",
  },
  "version-1.0.1-hello": Object {
    "category": "Guides",
    "id": "version-1.0.1-hello",
    "language": undefined,
    "localized_id": "version-1.0.1-hello",
    "permalink": "/docs/hello",
    "previous": "version-1.0.1-foo/baz",
    "previous_id": "version-1.0.1-foo/baz",
    "previous_title": "Baz",
    "sidebar": "version-1.0.1-docs",
    "source": "@versioned_docs/version-1.0.1/hello.md",
    "title": "Hello, World !",
    "version": "1.0.1",
  },
}
`;
